### ISSUE TITLE
Issue regarding threading with scrapers

### ISSUE THREAD

**Issue by AlphaBetrium** [itsToggle/plex_debrid#263]
Hi I've been experiencing an issue with the threading from the scrapers. It looks like a new thread is created for each new scraper call. I'm using Jackett along with this and it seems to be creating a thread for each indexer present in there as well. Would it be possible to add a single thread execution switch in the config or a thread pool? Cheers

**Comment by itsToggle** [comment]
Hi! here's some background info first:

plex_debrid is indeed creating a new thread for each scraper that you add in "/Settings/Scraper Settings/Sources/", so that they can all be queried in parallel.
This means that there is only a single thread created for the "jackett" scraper, not one for every indexers you add in jackett. However, some Jackett indexers don't provide torrent files or magnet links directly. They rather give a link that then leads to the actual torrent/magnet. plex_debrid therefore starts new threads for each release that needs to be resolved in this way, so they can all be resolved in parallel.

What exactly is your issue with the way these threads are created? How would a thread pool resolve them? :) I haven't worked with thread pools before, but I'm open to changing the way this script multi-threads if there are any benefits.

**Comment by AlphaBetrium** [comment]
Hey just want to say the work you've done is very impressive! I've completely changed my setup since I stumbled across plex_debrid along with your rclone fork. 

So the issue I've had is that too many threads are being created. I saw the thread count hit around 4500 and then the client recieved a runtime error as it couldn't create any more - I think that this is because of a limit on the number of threads a process is allowed to have. 

I initially removed the threading from the scraper file but ran into the same issue in the jackett service file - I was wrong the threads weren't per indexer but per result that was being resolved. 

I think a thread pool could come in handy for the scraper - if a pool of threads were to be created according to the length of the number of scrapers those threads could then be reused instead of creating new ones. 
For the threads created in the jackett resolving, I'm not too sure, they will surely vary in length for each torrent. Perhaps a max number of threads could be determined for this?

Thanks :)
